stackslib/src/net/connection.rs

```diff
@@ -1789,31 +1783,28 @@ mod test {
                     // in the foreground, get the messages
                     let nr = match conn.recv_data(&mut read) {
                         Ok(cnt) => cnt,
-                        Err(e) => match e {
-                            net_error::PermanentlyDrained => {
+                        Err(e) => {
+                            if let net_error::PermanentlyDrained = e {
```

Comment:
```suggestion
                        Err(net_error::PermanentlyDrained) => {
```

Is there a Clippy lint that will do this automatically?

Reply:
Not that I know of. Very possible though.